packages/paste-core/components/data-grid/__tests__/index.spec.tsx

```diff
@@ -42,6 +45,24 @@ describe('Data Grid', () => {
     });
   });
 
+  describe('Column Span', () => {
+    it('applies colSpan attribute as expected', () => {
+      const {getByTestId} = render(<ColumnSpanDataGrid />);
+      const th = getByTestId('data-grid-header');
+      expect(th).toHaveAttribute('colspan', '5');
+    });
+
+    it('enforces numeric colSpan prop type', () => {
```

Comment:
Issue: we don't usually test if prop types works as expected.... I'm also not sure this is test is correct because there isn't an expect. I think it would be ok to remove this test.

Reply:
It does raise the question, should we throw? If you're not using Typescript, theoretically you can pass a string to the html element, nothing stops you.

Reply:
Isn't that what the old school react prop types are for?

Reply:
Yeah, you can make it noisy (see my other comment about adding an old school proptype), but as a consumer you can just ignore them. It's not fail proof.

Reply:
I think it makes sense to throw and follows our patterns for props in other components